### PR TITLE
Add netstandard1.0 support

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change ###

Multi-targeting Xamarin.Forms.Core, Xamarin.Forms.Xaml, Xamarin.Forms.Maps to netstandard 1.0 so PCL users can use that without upgrading to netstandard 2.0.

- Cleanup sln by removing old configurations that we don't use.
- Remove confuse script about docs.
- Update nuspec

### Bugs Fixed ###

- Provide links to bugs here

### API Changes ###

Adding back for Netstandard1.0 only 

```
Xamarin.Forms.Internals.FileMode; 
Xamarin.Forms.Internals.FileAccess; 
Xamarin.Forms.Internals.FileShare; 
```

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Core/ImageSource.cs [hunk 1]
```diff
@@ -68,7 +67,19 @@ public static ImageSource FromResource(string resource, Type resolvingType)
 
 		public static ImageSource FromResource(string resource, Assembly sourceAssembly = null)
 		{
-			sourceAssembly = sourceAssembly ?? Assembly.GetCallingAssembly();
+			if (sourceAssembly == null)
+			{
+				MethodInfo callingAssemblyMethod = typeof(Assembly).GetTypeInfo().GetDeclaredMethod("GetCallingAssembly");
+				if (callingAssemblyMethod != null)
+				{
+					sourceAssembly = (Assembly)callingAssemblyMethod.Invoke(null, new object[0]);
```
Comment (WonyoungChoi) [inline Xamarin.Forms.Core/ImageSource.cs]: This reflection will be a problem when use .netcore runtime. (see https://bugzilla.xamarin.com/show_bug.cgi?id=60731).

Cause:
In .NET standard, the Assembly.GetCallingAssembly() has a [System.Security.DynamicSecurityMethod] attribute set (https://github.com/dotnet/coreclr/blob/13e7c4368da664a8b50228b1a5ef01a660fbb2dd/src/mscorlib/src/System/Reflection/Assembly.CoreCLR.cs#L248), which means callingAssemblyMethod.Attributes contains MethodAttributes.RequireSecObject flag, which in turn means that it cannot be invoked: (https://github.com/dotnet/coreclr/blob/97c58ac4fce27b7796206a59eea0ca27cb49fe1a/src/mscorlib/src/System/Reflection/RuntimeMethodInfo.cs#L48).

Comment (rmarinho) [reply Xamarin.Forms.Core/ImageSource.cs]: do you think this will work? netstandard2.0 should be the prefer way

**Comment by BradChase2011** [comment]
@rmarinho Is this something that will be making it into master soon?  We unfortunately cannot push our UWP to the store any longer because of the usual .net native tooling problems.  We have been releasing just fine under non < 2.0.  The .net native team havent responded to emails so we are really stuck at this moment needing to put the next version in the store but 2.0 only supports fall creators update and higher AND .Net Native tooling is broken.

File: Xamarin.Forms.Core/ImageSource.cs (hunk 1 above)
Comment (WonyoungChoi) [reply Xamarin.Forms.Core/ImageSource.cs]: yes it looks good :) thank you.

**Comment by rmarinho** [comment]
@BradChase2011 i m having build problems with this. still didn't figure why exactly. I will try to look at it again.

File: Xamarin.Forms.Platform.UAP/WindowsIsolatedStorage.cs [hunk 2]
```diff
@@ -4,13 +4,15 @@
 using Windows.Storage;
 using Windows.Storage.FileProperties;
 using Windows.Storage.Streams;
-using Xamarin.Forms.Internals;
+//using FileMode = Xamarin.Forms.Internals.FileMode;
```
Comment (samhouts) [inline Xamarin.Forms.Platform.UAP/WindowsIsolatedStorage.cs]: Do we need these here?

File: Xamarin.Forms.Platform.UAP/WindowsIsolatedStorage.cs [hunk 3]
```diff
@@ -103,7 +105,7 @@ public Task CreateDirectoryAsync(string path)
 
 				default:
 					throw new ArgumentException("access was an invalid FileAccess", "access");
-			}
+			}
```
Comment (samhouts) [inline Xamarin.Forms.Platform.UAP/WindowsIsolatedStorage.cs]: 👮

File: Xamarin.Forms.Platform.UAP/WindowsIsolatedStorage.cs (hunk 2 above)
Comment (rmarinho) [reply Xamarin.Forms.Platform.UAP/WindowsIsolatedStorage.cs]: Yes, we will try to get Netstandard1_0 after on UAP and will need this

File: Xamarin.Forms.Core/VisualElement_StyleSheet.cs [hunk 4]
```diff
@@ -28,11 +28,15 @@ BindableProperty IStylable.GetProperty(string key, bool inheriting)
...
 			var propertyOwnerType = styleAttribute.PropertyOwnerType ?? GetType();
+#if NETSTANDARD1_0
+			var bpField = propertyOwnerType.GetField(styleAttribute.BindablePropertyName);
+#else
 			var bpField = propertyOwnerType.GetField(styleAttribute.BindablePropertyName,
-													   BindingFlags.Public
-													 | BindingFlags.NonPublic
-													 | BindingFlags.Static
-													 | BindingFlags.FlattenHierarchy);
+													BindingFlags.Public
```
Comment (jassmith) [inline Xamarin.Forms.Core/VisualElement_StyleSheet.cs]: Whitespace issue

**Comment by StephaneDelcroix** [comment]
this PR should with this flag as well https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Core/Xaml/TypeConversionExtensions.cs#L197

(or rename the flag so it matches yours)

File: Xamarin.Forms.Core/Internals/Legacy/FileShare.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+	[EditorBrowsable(EditorBrowsableState.Never)]
+	public enum FileShare
+	{
+		None = 0,
+		Read = 1,
+		Write = 2,
+		ReadWrite = 3,
+		Delete = 4,
+		Inheritable = 16
+	}
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Internals/Legacy/FileShare.cs]:
```
None = 0,
Read = 1 << 0,
Write = 1 << 1.
ReadWrite = Read | Write,
Delete = 1 << 2,
Inheritable = 1 << 4,

File: Xamarin.Forms.Core/VisualElement.cs [hunk 6]
```diff
@@ -712,7 +712,9 @@ void IVisualElementController.InvalidateMeasure(InvalidationTrigger trigger)
 
 		internal void MockBounds(Rectangle bounds)
 		{
+#if NETSTANDARD2_0
 			(_mockX, _mockY, _mockWidth, _mockHeight) = bounds;
+#endif
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/VisualElement.cs]: missing an `#else`

File: Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+using System.ComponentModel;
+
+namespace Xamarin.Forms.Internals
+{
+	[EditorBrowsable(EditorBrowsableState.Never)]
+	public enum FileAccess
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs]: are those enums used only for netstandard1.0 ? or everywhere ?

File: Xamarin.Forms.Platform.UAP/WindowsIsolatedStorage.cs [hunk 8]
```diff
@@ -4,13 +4,15 @@
 using Windows.Storage;
 using Windows.Storage.FileProperties;
 using Windows.Storage.Streams;
-using Xamarin.Forms.Internals;
+//using FileMode = Xamarin.Forms.Internals.FileMode;
+//using FileAccess = Xamarin.Forms.Internals.FileAccess;
+//using FileShare = Xamarin.Forms.Internals.FileShare;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.UAP/WindowsIsolatedStorage.cs]: why do we need commented out `using`s ?

File: Xamarin.Forms.Core/FlexEnums.cs [hunk 9]
```diff
@@ -212,7 +212,7 @@ public override object ConvertFromInvariantString(string value)
 					if (value.Equals("auto", StringComparison.OrdinalIgnoreCase))
 						return Auto;
 					value = value.Trim();
-					if (value.EndsWith("%", StringComparison.InvariantCulture) && float.TryParse(value.Substring(0, value.Length - 1), NumberStyles.Number, CultureInfo.InvariantCulture, out float relflex))
+					if (value.EndsWith("%", StringComparison.OrdinalIgnoreCase) && float.TryParse(value.Substring(0, value.Length - 1), NumberStyles.Number, CultureInfo.InvariantCulture, out float relflex))
```
[the remark this answers is not in the document]
Comment (rmarinho) [reply Xamarin.Forms.Core/FlexEnums.cs]: nop

File: Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs (hunk 7 above)
Comment (rmarinho) [reply Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs]: only netstandard1.0

File: Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+	{
+		Read = 0x00000001,
```
[the remark this answers is not in the document]
Comment (rmarinho) [reply Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs]: i just copy what we wad before, this was part of Forms in previous versions

File: Xamarin.Forms.Platform.UAP/WindowsIsolatedStorage.cs (hunk 8 above)
Comment (rmarinho) [reply Xamarin.Forms.Platform.UAP/WindowsIsolatedStorage.cs]: i m going to add this later we need to make UAP compile under multi target for pre fall's update.  But thats another task

File: Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (rmarinho) [reply Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs]: we didn't had before.

Comment (StephaneDelcroix) [reply Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs]: so wrap them in `#if/#endif` please

[the remark this answers is not in the document]
Comment (StephaneDelcroix) [reply Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs]: add it then :) it was missing

Comment (rmarinho) [reply Xamarin.Forms.Core/Internals/Legacy/FileAccess.cs]: that's kinda of done my the csproj in a more clean way imo

**Pull request merged**